base/sparse/sparsematrix.jl

```diff
@@ -2252,6 +2252,8 @@ function setindex!{Tv}(A::SparseMatrixCSC{Tv}, x::Number,
     if isempty(I) || isempty(J); return A; end
     if !issorted(I); I = sort(I); end
     if !issorted(J); J = sort(J); end
+    if !allunique(I); I = unique(I); end
```

Comment:
I'd put this before the `!issorted` checks and use `sort!(I)` instead of `I = sort(I)`